From 36f047f39ade27757694eeec10d425e49eb2446a Mon Sep 17 00:00:00 2001
From: Steve Dickson <SteveD@redhat.com>
Date: Thu, 18 Sep 2014 14:19:37 -0400
Subject: [fs] nfs: check hostname in nfs_get_client

Message-id: <1411050057-12417-15-git-send-email-steved@redhat.com>
Patchwork-id: 96182
O-Subject: [RHEL7.1 PATCH 14/94] nfs: check hostname in nfs_get_client
Bugzilla: 1111708
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>

From: Peng Tao <tao.peng@primarydata.com>

We reference cl_hostname in many places. Add a check to make
sure it exists.

Signed-off-by: Peng Tao <tao.peng@primarydata.com>
Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Upstream Id: 31434f496abb9f3410b10f541462fe58613dd3ad
Bugzilla: 1111708
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfs/client.c b/fs/nfs/client.c
index 1d09289..b213ee8 100644
--- a/fs/nfs/client.c
+++ b/fs/nfs/client.c
@@ -482,8 +482,13 @@ nfs_get_client(const struct nfs_client_initdata *cl_init,
  struct nfs_net *nn = net_generic(cl_init->net, nfs_net_id);
  const struct nfs_rpc_ops *rpc_ops = cl_init->nfs_mod->rpc_ops;
 
+ if (cl_init->hostname == NULL) {
+  WARN_ON(1);
+  return NULL;
+ }
+
  dprintk("--> nfs_get_client(%s,v%u)\n",
-  cl_init->hostname ?: "", rpc_ops->version);
+  cl_init->hostname, rpc_ops->version);
 
  /* see if the client already exists */
  do {
@@ -510,7 +515,7 @@ nfs_get_client(const struct nfs_client_initdata *cl_init,
  } while (!IS_ERR(new));
 
  dprintk("<-- nfs_get_client() Failed to find %s (%ld)\n",
-  cl_init->hostname ?: "", PTR_ERR(new));
+  cl_init->hostname, PTR_ERR(new));
  return new;
 }
 EXPORT_SYMBOL_GPL(nfs_get_client);
-- 
1.7.1